### PR TITLE
Support requestBody

### PR DESCRIPTION
**Pull request by ayman2012**
Add a new property for request builder to support request body
- headers for request header
- parameters for url query params
- bodyParams for body params

### CODE REVIEW

File: NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift [hunk 1]
```diff
@@ -10,11 +10,11 @@ import Foundation
 
 public protocol RequestBuilder {
     var baseURL: String { get }
-
     var method: HttpMethod { get }
     var path: String { get }
     var parameters: [String: Any] { get }
     var headers: [String: String] { get }
+    var bodyParamters: [String: Any] {get}
```
Comment (abozaid-ibrahim) [inline NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift]: remove this, and  use parameters var to do the same job

File: NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift [hunk 2]
```diff
@@ -43,6 +43,13 @@ extension RequestBuilder {
         myURL?.queryItems = items
         var request = URLRequest(url: myURL!.url!, cachePolicy: URLRequest.CachePolicy.reloadIgnoringCacheData, timeoutInterval: APIConstants.timeout)
         request.httpMethod = method.rawValue
+        switch method {
```
Comment (abozaid-ibrahim) [inline NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift:43]: switch method {
        case .get:
           myURL?.queryItems = items
        var request = URLRequest(url: myURL!.url!, cachePolicy: URLRequest.CachePolicy.reloadIgnoringCacheData, timeoutInterval: APIConstants.timeout)
        request.httpMethod = method.rawValue

        default:
/// here use 
            let jsonData = try? JSONSerialization.data(withJSONObject: `parameters`)
            request.httpBody = jsonData
        }
        request.allHTTPHeaderFields = request.allHTTPHeaderFields?.merging(headers, uniquingKeysWith: { (_, newK) -> String in
           
        })

File: NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/example/XEndpoint.swift [hunk 3]
```diff
@@ -11,6 +11,12 @@ enum XEndpoint {
 }
 
 extension XEndpoint: RequestBuilder {
+  
+
+    var bodyParamters: [String : Any] {
```
Comment (abozaid-ibrahim) [inline NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/example/XEndpoint.swift:16]: it's duplication for the params, the user will get confused

**Review by abozaid-ibrahim — COMMENTED** [review]
fix the comments

File: NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift (hunk 1 above)
Comment (ayman2012) [reply NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift]: Done

File: NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift (hunk 2 above)
Comment (ayman2012) [reply NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/RequestBuilder.swift:43]: Done

File: NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/example/XEndpoint.swift (hunk 3 above)
Comment (ayman2012) [reply NetworkPackageDemo/NetworkPackageDemo/Sources/NetworkPackage/example/XEndpoint.swift:16]: Done

**Review by abozaid-ibrahim — APPROVED** [review]

**Pull request merged**